blockchain/blocks/src/tipset.rs

```diff
@@ -41,6 +41,13 @@ pub struct Tipset {
     key: OnceCell<TipsetKeys>,
 }
 
+impl From<&BlockHeader> for Tipset {
+    fn from(value: &BlockHeader) -> Self {
+        //unfallible since there's just one BlockHeader
+        Tipset::new(vec![value.clone()]).unwrap()
```

Comment:
I'd prefer if you inlined `Tipset::new`. The code is quite short.

Reply:
And you can also write an implementation for `BlockHeader` (ie when not using a reference).

Reply:
Done